### PR TITLE
add doc for dynamic config

### PR DESCRIPTION
**Pull request by ran-huang**
### What is changed, added or deleted? (Required)

Add a doc for dynamic config.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

**If you select two or more versions from above**, to trigger the bot to cherry-pick this PR to your desired release version branch(es), you **must** add corresponding labels such as **needs-cherry-pick-4.0**, **needs-cherry-pick-3.1**, **needs-cherry-pick-3.0**, and **needs-cherry-pick-2.1**.

### What is the related PR or file link(s)?

- This PR is translated from:https://github.com/pingcap/docs-cn/pull/3058
- Other reference link(s):

### CODE REVIEW

File: dynamic-config.md [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+| replication.enable-placement-rules | Enables placement rules |
+| replication.strictly-match-label | Enables the label check |
+| pd-server.use-region-storage | Enables independent Region storage |
+| pd-server.max-gap-reset-ts | Sets the maximum interval of resetting timestamp (BR) |
+| pd-server.key-type| Sets the cluster key type |
+| pd-server.metric-storage | Sets the storage address of the cluster metrics |
+| pd-server.dashboard-address | Sets the dashboard address |
+| replication-mode.replication-mode | Sets the backup mode |
+
+For detailed parameter description, refer to [PD Configuration File](https://pingcap.com/docs/stable/pd-configuration-file/).
```
Comment (ran-huang) [inline dynamic-config.md:136]:
```suggestion
For detailed parameter description, refer to [PD Configuration File](/pd-configuration-file.md).
```

File: dynamic-config.md [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+| {db-name}.{cf-name}.max-compaction-bytes | The maximum number of bytes written into disk per compaction |
+| {db-name}.{cf-name}.max-bytes-for-level-multiplier | The default amplification multiple for each layer |
+| {db-name}.{cf-name}.disable-auto-compactions | Enables or disables automatic compaction |
+| {db-name}.{cf-name}.soft-pending-compaction-bytes-limit | The soft limit on the pending compaction bytes |
+| {db-name}.{cf-name}.hard-pending-compaction-bytes-limit | The hard limit on the pending compaction bytes |
+| {db-name}.{cf-name}.titan.blob-run-mode | The mode of processing blob files |
+
+In the table above, parameters with the `{db-name}` or `{db-name}.{cf-name}` prefix is configurations related to RocksDB. The value of `db-name` can be `rocksdb` or `raftdb`.
+
+- When `db-name` is `rocksdb`, the value of `cf-name` can be `defaultcf`, `writecf`, `lockcf`, and `raftcf`.
```
[the remark this answers is not in the document]
Comment (rleungx) [reply dynamic-config.md:209]: It's correct.

File: dynamic-config.md [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+
+## Common Operations
+
+This section describes the common operations of dynamic configuration change.
+
+### View instance configuration
+
+To view the configuration of all instances in the cluster, use the `show config` SQL statement. The result is as follows:
+
+```
```
[the remark this answers is not in the document]
Comment (rleungx) [reply dynamic-config.md:27]: You are right.

[the remark this answers is not in the document]
Comment (ran-huang) [reply dynamic-config.md:27]: Thanks! I'll fix it later.

File: dynamic-config.md [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+
+This document describes how to use the dynamic configuration change feature.
+
+> **Note:**
+>
+> This feature is experimental. It is **NOT** recommended to use this feature in the production environment.
+
+The feature of dynamic configuration change is to online update the configuration of components (including TiKV and PD) using SQL statements.
+
+Dynamic configuration change cannot be used to update the TiDB configuration. If you want to change the behavior of TiDB, modify its corresponding SQL variables.
```
Comment (TomShawn) [inline dynamic-config.md:17]:
```suggestion
Dynamic configuration change cannot be used to update the TiDB configuration. If you want to change the behavior of TiDB, modify TiDB's corresponding SQL variables.
```

File: dynamic-config.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (TomShawn) [reply dynamic-config.md:27]: @ran-huang Please remove `mysql` in this document.

**Pull request merged**